cmd/agent/app/integrations.go

```diff
@@ -427,65 +340,108 @@ func install(cmd *cobra.Command, args []string) error {
 		)
 	}
 
-	// Run pip check first to see if the python environment is clean
-	if err := pipCheck(cachePath); err != nil {
+	// Download the wheel
+	wheelPath, err := downloadWheel(integration, versionToInstall.String())
+	if err != nil {
+		return fmt.Errorf("error when downloading the wheel for %s %s: %v", integration, versionToInstall, err)
+	}
+
+	// Verify datadog_checks_base is compatible with the requirements
+	shippedBaseVersion, err := installedVersion("datadog-checks-base")
+	if err != nil {
+		return fmt.Errorf("unable to get the version of datadog_checks_base: %v", err)
+	}
+	if ok, err := validateBaseDependency(wheelPath, shippedBaseVersion); err != nil {
+		return fmt.Errorf("unable to validate compatibility of %s with the agent: %v", wheelPath, err)
+	} else if !ok {
 		return fmt.Errorf(
-			"error when validating the agent's python environment, won't install %s: %v",
-			integration, err,
+			"%s %s is not compatible with datadog_checks_base %s shipped in the agent",
+			integration, versionToInstall, shippedBaseVersion,
 		)
 	}
 
 	// Install the wheel
-	if err := pip(append(pipArgs, args[0]), false); err != nil {
-		return err
+	if err := pip(append(pipArgs, wheelPath)); err != nil {
+		return fmt.Errorf("error installing wheel %s: %v", wheelPath, err)
 	}
 
-	// Run pip check to determine if the installed integration is compatible with the base check version
-	pipCheckErr := pipCheck(cachePath)
-	if pipCheckErr == nil {
-		err := moveConfigurationFilesOf(integration)
-		if err == nil {
-			fmt.Println(color.GreenString(fmt.Sprintf(
-				"Successfully installed %s %s", integration, versionToInstall,
-			)))
-			return nil
-		}
-
+	// Move configuration files
+	if err := moveConfigurationFilesOf(integration); err != nil {
 		fmt.Printf("Installed %s %s", integration, versionToInstall)
 		return fmt.Errorf("Some errors prevented moving %s configuration files: %v", integration, err)
 	}
 
-	// We either detected a mismatch, or we failed to run pip check
-	// Either way, roll back the install and return the error
-	if currentVersion == nil {
-		// Special case where we tried to install a new integration, not yet released with the agent
-		pipArgs = []string{
-			"uninstall",
-			integration,
-			"-y",
-		}
-	} else {
-		pipArgs = append(pipArgs, fmt.Sprintf("%s==%s", integration, currentVersion))
+	fmt.Println(color.GreenString(fmt.Sprintf(
+		"Successfully installed %s %s", integration, versionToInstall,
+	)))
+	return nil
+}
+
+func downloadWheel(integration, version string) (string, error) {
+	pyPath, err := getCommandPython()
+	if err != nil {
+		return "", err
+	}
+	args := []string{
+		"-m", "datadog_checks.downloader",
```

Comment:
looks like we're not calling the downloader executable directly, so the constant we've defined `downloaderBin` I _think_ is useless. Maybe we should call it `downloaderModule = "datadog_checks.downloader"` and use that here.